src/k8s/cmd/k8s/k8s_bootstrap_test.go

```diff
@@ -0,0 +1,92 @@
+package k8s
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	apiv1 "github.com/canonical/k8s/api/v1"
+	. "github.com/onsi/gomega"
+)
+
+type testCase struct {
+	name           string
+	yamlConfig     string
+	expectedConfig apiv1.BootstrapConfig
+	expectedError  string
+}
+
+var testCases = []testCase{
+	{
+		name: "CompleteConfig",
+		yamlConfig: `
+components:
+  - network
+  - dns
+  - gateway
+  - ingress
+  - storage
+  - metrics-server
+cluster-cidr: "10.244.0.0/16"
+enable-rbac: true
+k8s-dqlite-port: 12379`,
+		expectedConfig: apiv1.BootstrapConfig{
+			Components:    []string{"network", "dns", "gateway", "ingress", "storage", "metrics-server"},
+			ClusterCIDR:   "10.244.0.0/16",
+			EnableRBAC:    &[]bool{true}[0],
```

Comment:
nit: you can use the `vals.Pointer(true)` helper here (`pkg/utils/vals/pointer.go`)

Reply:
Ah, much better! Thanks :)